File: benchmarks/testdata/sample.ts
/**
 * Sample TypeScript file for testing pattern detection
 */

const API_KEY = "AKIAIOSFODNN7EXAMPLE";
const STRIPE_KEY = "sk_live_51HdY8eZvKYlo2Cw3jHGmXj";
const GITHUB_TOKEN = "ghp_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx";

interface Config {
  database: {
    host: string;
    port: number;
    username: string;
    password: string;
  };
  aws: {
    accessKeyId: string;
    secretAccessKey: string;
  };
}

const config: Config = {
  database: {
    host: "localhost",
    port: 5432,
    username: "admin",
    password: "secretpassword123",
  },
  aws: {
    accessKeyId: "AKIAIOSFODNN7EXAMPLE",
    secretAccessKey: "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
  },
};

async function fetchData(apiKey: string): Promise<void> {
  const response = await fetch("https://api.example.com/data", {
    headers: {
      "Authorization": `Bearer ${apiKey}`,
      "X-API-Key": API_KEY,
    },
  });
  console.log("Data fetched successfully");
}

export async function processData(): Promise<void> {
  await fetchData(API_KEY);
  console.log("Processing complete");
}

File: benchmarks/testdata/sample.py
#!/usr/bin/env python3
"""
Sample Python file for testing pattern detection.
"""

import os
import sys
import logging
from typing import Optional

# Configuration
API_KEY = "AKIAIOSFODNN7EXAMPLE"
SECRET_TOKEN = "ghp_xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
DATABASE_URL = "postgresql://admin:password123@localhost:5432/mydb"

def process_data(data: str) -> dict:
    """Process incoming data with API integration."""
    logger.info(f"Processing data with API key: {API_KEY[:8]}...")
    headers = {
        "Authorization": f"Bearer {SECRET_TOKEN}",
        "X-API-Key": API_KEY,
    }
    return {"status": "success", "records": len(data)}

def connect_to_database():
    """Connect to database with credentials."""
    connection_string = f"postgresql://admin:{DATABASE_URL.split(':')[2].split('@')[0]}@localhost:5432/mydb"
    print(f"Connecting to database...")

class DataProcessor:
    def __init__(self, api_key: str):
        self.api_key = api_key
        self.stripe_key = "sk_live_51HdY8eZvKYlo2Cw3jHGmXj"

    def process(self):
        pass

if __name__ == "__main__":
    processor = DataProcessor(API_KEY)
    result = process_data("test data")
    print(result)
